app/models/host/managed.rb

```diff
@@ -985,7 +995,7 @@ def ensure_associations
   end
 
   def set_default_user
-    return if self.owner_type.present? && !OWNER_TYPES.include?(self.owner_type)
+    return if self.owner_type.present? && !OWNER_TYPES.include?(self.owner_type) # to propagate invalid value to validations
     self.owner_type ||= 'User'
```

Comment:
In relation to https://github.com/theforeman/foreman/pull/3740, I guess we should set the default user only when the `owner_type` is not specified explicitly